launch/tests/next_schedule_test.rs

```diff
@@ -117,52 +184,338 @@ fn test_no_pickup_dropoff(fixture_model: BaseModel) -> Result<(), Error> {
+        assert_eq!(result.len(), 4);
+
+        // calendar.start  < calendar.end < from_datetime < until_datetime
+        config.from_datetime = "2020-01-10T10:00:00".as_datetime();
+        config.duration = 30 * 24 * 60 * 60; // 1 month
+        let result = build_and_solve_schedule(&config, &fixture_model);
+        let error = format!("{:?}", result.as_ref().err().unwrap());
+        assert_eq!(error, "BadFromDatetime");
+    }
+
+    Ok(())
+}
+
+#[rstest]
+fn test_past_midnight_vehicle(fixture_model: BaseModel) -> Result<(), Error> {
+    let _log_guard = launch::logger::init_test_logger();
+
+    // This code test that vehicle valid on day : from_date - 2 & from_date -1
+    // and with a next_schedule (departure/arrival) after from_datetime are returned as expected
+
+    // Departure Tests
+    {
+        let mut config = ScheduleConfig::new(
+            ScheduleOn::BoardTimes,
+            "stop_area:sa:J",
+            "2020-01-02T00:00:00",
+        );
+        config.duration = 10 * 60; // 10m
+        let result = build_and_solve_schedule(&config, &fixture_model)?;
+        assert_eq!(result.len(), 1);
+        let stop_time = &result[0];
+        assert_eq!(stop_time.time, "2020-01-02T00:10:00".as_datetime());
+
+        config.duration = 30 * 60; // 30m
+        let result = build_and_solve_schedule(&config, &fixture_model)?;
+        assert_eq!(result.len(), 2);
+        let stop_time = &result[1];
+        assert_eq!(stop_time.time, "2020-01-02T00:30:00".as_datetime());
+    }
+
+    // Arrival Tests
+    {
+        let mut config = ScheduleConfig::new(
+            ScheduleOn::DebarkTimes,
+            "stop_area:sa:J",
+            "2020-01-02T00:00:00",
+        );
+        config.duration = 10 * 60; // 10m
+        let result = build_and_solve_schedule(&config, &fixture_model)?;
+        assert_eq!(result.len(), 1);
+        let stop_time = &result[0];
+        assert_eq!(stop_time.time, "2020-01-02T00:10:00".as_datetime());
+
+        config.duration = 30 * 60; // 30m
+        let result = build_and_solve_schedule(&config, &fixture_model)?;
+        assert_eq!(result.len(), 2);
+        let stop_time = &result[1];
+        assert_eq!(stop_time.time, "2020-01-02T00:30:00".as_datetime());
+    }
+
+    Ok(())
+}
+
+#[rstest]
+fn test_on_route(fixture_model: BaseModel) -> Result<(), Error> {
+    let _log_guard = launch::logger::init_test_logger();
+
+    // In this test we expect to receive all departure of all stop_points of route R1
+    // after a certain datetime (we test multiple datetime)
+
+    // Departure Tests
+    {
+        let mut config =
+            ScheduleConfig::new(ScheduleOn::BoardTimes, "route:R1", "2020-01-01T10:00:00");
+        config.duration = 10 * 60 * 60; // 10h
+
+        // on route R1 we have 9 stop_points and 6 valid vj per day
+        // we expect 6 Next Departure per day with from_datetime = 2020-01-02T10:00:00
```

Comment:
```suggestion
        // we expect 4 Next Departure with from_datetime = 2020-01-02T10:00:00
```
Could you explain in the comment why you expect 4 results ?